docs/docs/forms.md

```diff
@@ -335,6 +335,47 @@ For example:
 ```
 will return `undefined` if the field is empty.
 
+### Custom Input Fields
+
+You can create a custom field that integrates with Redwood through the use of `useRegister` and `useErrorStyles`. Each of these serving a different purpose depending on what you are trying to build.
+
+`useRegister` registers the field with react-hook-form and is a wrapper for [`register`](https://react-hook-form.com/api/useform/register).
+
+`useErrorStyles` sets up error styling for your custom input field.
+
+Using these two together you can create custom input fields that replicate a redwood input field while also allowing for custom domain logic.
+
+In the following example we have an all-in-one custom required input field with label, input, and error display.
+
+```jsx
+import { FieldError, useErrorStyles, useRegister } from '@redwoodjs/forms'
+
+const RequiredField = ({label, name, validation}) => {
+  const register = useRegister({
+    name,
+    validation: {...validation, required: true}
+  })
+
+  const labelStyles = useErrorStyles({
+    className: `my-label-class`,
+    errorClassName: `my-label-error-class`,
+    name,
+  })
+
+  const inputStyles = useErrorStyles({
+    className: `my-input-class`,
+    errorClassName: `my-input-error-class`,
+    name,
+  })
+
+  return (
+    <label {...labelStyles}>{label}</label>
```

Comment:
I'm not a big fan of spreading props. I like to be more explicit with what props I pass to a component. What does `useErrorStyles()` return? If it's not too much work, could you destructure the returned object and pass each part as a separate prop?